packages/eslint-plugin/rules/valid-sprintf.js

```diff
@@ -70,44 +87,64 @@ module.exports = {
 				}
 
 				if ( ! candidates.length ) {
-					context.report(
+					context.report( {
 						node,
-						'sprintf must be called with a valid format string'
-					);
+						messageId: 'invalidFormatString',
+					} );
 					return;
 				}
 
 				let numPlaceholders;
-				for ( let i = 0; i < candidates.length; i++ ) {
-					const match = candidates[ i ].match( REGEXP_PLACEHOLDER );
+				for ( const candidate of candidates ) {
+					const allMatches = candidate.match(
+						SPRINTF_PLACEHOLDER_REGEX
+					);
+					const unorderedMatches = candidate.match(
+						UNORDERED_SPRINTF_PLACEHOLDER_REGEX
+					);
```

Comment:
I know we'd talked about named placeholders recently, but: For better or worse, `wp.i18n.sprintf` supports them via `sprintf-js` ([example](https://runkit.com/aduth/5e6955dc403b0c0013a96a93)). I think this rule would effectively forbid them? I could understand this being desirable in context of WordPress and how those strings couldn't be reused in PHP (bad), just wondering if it's something we want to be the default for this rule generally.

Reply:
Hmm :/ Interesting.

I suppose the regex would have to be updated for this to work..

Reply:
From [my earlier comment](https://github.com/WordPress/gutenberg/pull/20574#discussion_r391271540), this one could work:

https://github.com/aduth/tannin/blob/cd1c7447843df7751c4abd1b92aee03fe56bfba4/packages/sprintf/index.js#L32

It matches named arguments (https://github.com/aduth/tannin/pull/1), and has been battle-tested in WordPress.com Calypso (https://github.com/Automattic/wp-calypso/pull/34845).

Reply:
Updated the regex now and it works like a charm 👍